components/core/src/clp/Grep.cpp

```diff
@@ -1063,4 +1044,468 @@ size_t Grep::search(Query const& query, size_t limit, Archive& archive, File& co
+        bool has_proceeding_delimiter
+                = processed_search_string.size() == end_idx || is_greedy_wildcard[end_idx]
+                  || is_non_greedy_wildcard[end_idx]
+                  || (false == is_escape[end_idx]
+                      && lexer.is_delimiter(processed_search_string[end_idx]))
+                  || (is_escape[end_idx] && lexer.is_delimiter(processed_search_string[end_idx + 1])
+                  );
+
+        // If the substring contains a wildcard, we need to consider the case that it can
+        // simultaneously match multiple variables and static text, and we need a different
+        // approach to compare against the archive.
+        bool contains_wildcard = false;
+        set<uint32_t> variable_types;
+        if (has_preceding_delimiter && has_proceeding_delimiter) {
+            // If the substring is preceded or proceeded by a greedy wildcard then it's
+            // possible the substring could be extended to match a var, so the wildcards are
+            // added to the substring. If we don't consider this case we could miss
+            // combinations. Take for example "a*b", "a*" and "*b" can both match a has#
+            // style variable ("\w*\d+\w*"). If we decompose the string into either
+            // substrings "a*" + "b" or "a" + "*b", neither would capture the possibility of
+            // a logtype with the form "<has#>*<has#>", which is a valid possibility during
+            // compression. Instead we desire to decompose the string into "a*" + "*" +
+            // "*b". Note, non-greedy wildcards do not need to be considered, for example
+            // "a?b" can never match "<has#>?<has#>" or "<has#><has#>".
+            uint32_t substr_start = begin_idx;
+            uint32_t substr_end = end_idx;
+            bool prev_char_is_star = begin_idx > 0 && is_greedy_wildcard[begin_idx - 1];
+            bool next_char_is_greedy_wildcard
+                    = end_idx < processed_search_string.length() && is_greedy_wildcard[end_idx];
+            if (prev_char_is_star) {
+                substr_start--;
+            }
+            if (next_char_is_greedy_wildcard) {
+                substr_end++;
+            }
+            std::tie(variable_types, contains_wildcard) = get_substring_variable_types(
+                    string_view(processed_search_string)
+                            .substr(substr_start, substr_end - substr_start),
+                    substr_start,
+                    is_greedy_wildcard,
+                    is_non_greedy_wildcard,
+                    is_escape,
+                    lexer
+            );
+            bool already_added_var = false;
+            // Use the variable types to determine the possible_substr_types
+            for (int id : variable_types) {
+                auto& schema_type = lexer.m_id_symbol[id];
+                if (schema_type != "int" && schema_type != "float") {
```

Comment:
We should replace `"int"` and `"float"` with constants.

That said, in future PRs we should extend the schema file format to support some way of indicating how a variable type can be encoded (I think you worked on something like that a long time ago).